doc/manual/building_developing.rst

```diff
@@ -0,0 +1,172 @@
+
+- Be aware that Vivado is notoriously not a lightweight piece of software and you will likely need **at least 70GB+** of free space to install it. 
+- If you do not want to write to ``/opt``, you can install into a folder of your home directory. 
+- During the Vivado installation, uncheck ``Install cable drivers`` (they are not required, as we use better open source alternatives).
+- If the Vivado GUI installer crashes, you may be able to work around the problem by running it in unattended mode with a command such as ``./xsetup -a XilinxEULA,3rdPartyEULA,WebTalkTerms -b Install -e 'Vitis Unified Software Platform' -l /opt/Xilinx/``. 
+- Vivado installation issues are not uncommon. Searching for similar problems on `the M-Labs forum <https://forum.m-labs.hk/>`_ or `Vivado's own support forums <https://support.xilinx.com/s/topic/0TO2E000000YKXwWAO/installation-and-licensing>`_ might be helpful when looking for solutions. 
+
+
+.. _system-description: 
+
+System description file
+-----------------------
+
+ARTIQ gateware and firmware binaries are dependent on the system configuration. In other words, a specific set of ARTIQ binaries is bound to the exact arrangement of real-time hardware it was generated for: the core device itself, its role in a DRTIO context (master, satellite, or standalone), the (real-time) peripherals in use, the physical EEM ports they will be connected to, and various other basic specifications. This information is provided to the software in the form of a JSON file called the system description or system configuration file.
+
+If you already have your system configuration file on hand, you can edit it to reflect any changes in configuration. If you purchased your original system from M-Labs, or recently purchased new hardware to add to it, you can obtain your up-to-date system configuration file through AFWS at any time using the command ``$ afws_client get_json``. If you are starting from scratch, a close reading of ``coredevice_generic.schema.json`` in ``artiq/coredevice`` will be helpful.
+
+System descriptions do not need to be very complex. At its most basic, a system description looks something like: :: 
+
+    {
+        "target": "kasli",
+        "variant": "example",
+        "hw_rev": "v2.0",
+        "base": "master",
+        "peripherals": [
+            {
+                "type": "grabber",
+                "ports": [0]
+            }
+        ]
+    }
+
+Only these five fields are required, and the ``peripherals`` list can in principle be empty. A limited number of more extensive examples can currently be found in `the ARTIQ-Zynq repository <https://git.m-labs.hk/M-Labs/artiq-zynq/src/branch/master>`_. Once your system description file is complete, you can use ``artiq_ddb_template`` to test it and to generate a template for the corresponding :ref:`device database <device-db>`.  
```

Comment:
There's also one example in main repo, with shuttler: https://github.com/m-labs/artiq/blob/master/artiq/examples/kasli_shuttler/kasli_shuttler.json